umh-core/pkg/fsm/redpanda/machine.go

```diff
@@ -0,0 +1,160 @@
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package redpanda
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/looplab/fsm"
+
+	internal_fsm "github.com/united-manufacturing-hub/united-manufacturing-hub/umh-core/internal/fsm"
+	"github.com/united-manufacturing-hub/united-manufacturing-hub/umh-core/pkg/config"
+	"github.com/united-manufacturing-hub/united-manufacturing-hub/umh-core/pkg/logger"
+	"github.com/united-manufacturing-hub/united-manufacturing-hub/umh-core/pkg/metrics"
+	redpanda_service "github.com/united-manufacturing-hub/united-manufacturing-hub/umh-core/pkg/service/redpanda"
+)
+
+// NewRedpandaInstance creates a new RedpandaInstance with the given ID and service path
+func NewRedpandaInstance(
+	config config.RedpandaConfig) *RedpandaInstance {
+
+	cfg := internal_fsm.BaseFSMInstanceConfig{
+		ID:                           config.Name,
+		DesiredFSMState:              OperationalStateStopped,
+		OperationalStateAfterCreate:  OperationalStateStopped,
+		OperationalStateBeforeRemove: OperationalStateStopped,
+		OperationalTransitions: []fsm.EventDesc{
+			// Basic lifecycle transitions
+			// Stopped is the initial state
+			{Name: EventStart, Src: []string{OperationalStateStopped}, Dst: OperationalStateStarting},
+
+			// Starting phase transitions
+			{Name: EventS6Started, Src: []string{OperationalStateStarting}, Dst: OperationalStateStartingConfigLoading},
+			{Name: EventConfigLoaded, Src: []string{OperationalStateStartingConfigLoading}, Dst: OperationalStateStartingWaitingForHealthchecks},
+			{Name: EventHealthchecksPassed, Src: []string{OperationalStateStartingWaitingForHealthchecks}, Dst: OperationalStateStartingWaitingForServiceToRemainRunning},
+			{Name: EventStartDone, Src: []string{OperationalStateStartingWaitingForServiceToRemainRunning}, Dst: OperationalStateIdle},
+			{Name: EventStop, Src: []string{OperationalStateStarting, OperationalStateStartingConfigLoading, OperationalStateStartingWaitingForHealthchecks, OperationalStateStartingWaitingForServiceToRemainRunning}, Dst: OperationalStateStopping},
+
+			// From any starting state, we can either go back to OperationalStateStarting (e.g., if there was an error)
+			{Name: EventStartFailed, Src: []string{OperationalStateStarting, OperationalStateStartingConfigLoading, OperationalStateStartingWaitingForHealthchecks, OperationalStateStartingWaitingForServiceToRemainRunning}, Dst: OperationalStateStarting},
+
+			// Running phase transitions
+			// From Idle, we can go to Active when data is processed or to Stopping
```

Comment:
i assume idle does not make sense right? what are your thoughts on the potential states of redpanda?